opentracing-shim/src/main/java/openconsensus/opentracingshim/SpanBuilderShim.java

```diff
@@ -59,37 +68,94 @@ public SpanBuilder ignoreActiveSpan() {
 
   @Override
   public SpanBuilder withTag(String key, String value) {
-    // TODO
+    if ("span.kind".equals(key)) {
+      switch (value) {
+        case "CLIENT":
+          this.builder.setSpanKind(Kind.CLIENT);
+          break;
+        case "SERVER":
+          this.builder.setSpanKind(Kind.SERVER);
+          break;
+        case "PRODUCER":
+          this.builder.setSpanKind(Kind.PRODUCER);
+          break;
+        case "CONSUMER":
+          this.builder.setSpanKind(Kind.CONSUMER);
+          break;
+        default:
+          this.builder.setSpanKind(Kind.UNDEFINED);
+          break;
+      }
+    } else if ("error".equals(key)) {
+      // TODO: confirm we can ignore it
+      // https://github.com/bogdandrutu/openconsensus/issues/41
+    } else {
+      this.spanBuilderAttributes.put(key, AttributeValue.stringAttributeValue(value));
+    }
+
     return this;
   }
 
   @Override
   public SpanBuilder withTag(String key, boolean value) {
-    // TODO
+    if ("error".equals(key) && (value == true)) {
+      // TODO: confirm we can ignore it
+      // https://github.com/bogdandrutu/openconsensus/issues/41
+    } else {
+      this.spanBuilderAttributes.put(key, AttributeValue.booleanAttributeValue(value));
+    }
     return this;
   }
 
   @Override
   public SpanBuilder withTag(String key, Number value) {
-    // TODO
+    // TODO - Verify only the 'basic' types are supported/used.
+    if (value instanceof Integer
+        || value instanceof Long
+        || value instanceof Short
+        || value instanceof Byte) {
+      this.spanBuilderAttributes.put(key, AttributeValue.longAttributeValue(value.longValue()));
+    } else if (value instanceof Float || value instanceof Double) {
+      this.spanBuilderAttributes.put(key, AttributeValue.doubleAttributeValue(value.doubleValue()));
+    } else {
+      throw new IllegalArgumentException("Number type not supported");
+    }
+
     return this;
   }
 
   @Override
   public <T> SpanBuilder withTag(Tag<T> tag, T value) {
-    // TODO
+    if (value instanceof String) {
+      this.withTag(tag.getKey(), (String) value);
+    } else if (value instanceof Boolean) {
+      this.withTag(tag.getKey(), ((Boolean) value).booleanValue());
+    } else if (value instanceof Number) {
+      this.withTag(tag.getKey(), (Number) value);
+    } else {
+      this.withTag(tag.getKey(), value.toString());
+    }
+
     return this;
   }
 
   @Override
   public SpanBuilder withStartTimestamp(long microseconds) {
-    // TODO
+    this.spanBuilderAttributes.put("timestamp", AttributeValue.longAttributeValue(microseconds));
```

Comment:
should this be "start_timestamp" and equivalent for "end_timestamp"? Maybe consider to have ot specific tags like "ot.start_timestamp"?